libs/error-stack/src/ext/iter.rs

```diff
@@ -81,10 +81,22 @@ where
     report.map_or_else(|| Ok(value), |report| Err(report))
 }
 
-/// An extension trait for iterators that allows collecting items while handling errors.
+/// An extension trait for iterators that enables error-aware collection of items.
 ///
-/// This trait provides additional functionality to iterators that yield `Result` items,
-/// allowing them to be collected into a container while propagating any errors encountered.
+/// This trait enhances iterators yielding `Result` items by providing methods to
+/// collect successful items into a container while aggregating encountered errors.
+///
+/// # Performance Considerations
+///
+/// These methods may have performance implications as they potentially iterate
+/// through the entire collection, even after encountering errors.
+///
+/// # Unstable Feature
+///
+/// This trait is currently available only under the `unstable` feature flag and
+/// does not adhere to semver guarantees. Its API may change in future releases.
```

Comment:
I don't think we need to mention this everywhere as the documentation is properly generating this:
<img width="312" alt="image" src="https://github.com/user-attachments/assets/623065ba-cc23-47ea-9f44-b2aa2bff16c1">

Reply:
I just wanted to make sure to mention that it is exempt from semver guarantees and why, not necessarily that it's unstable.